### ISSUE TITLE
check_cpl_reel_coherence: Encryption is not required to be homogenous throughout composition

### ISSUE THREAD

**Issue by matmat** [Ymagis/ClairMeta#35]
Please relax the check on encryption coherence across reels with regards to encryption.

The below references do not mention encryption and indeed, I have several sample compositions containing a mix of encrypted and unencrypted trackfiles that playout without problems.

- ST 429-2:2013 section 8.7:
_Essence tracks in a Composition shall have homogenous encoding parameter values throughout the Composition. Picture essence shall have constant **frame rate** and **pixel array size**. Sound essence shall have constant **sample rate**, **language**, **channel count**, and **channel assignment parameters**._

- mpeg_ii_pack_constraints_2v5.doc section 7.2:
_The **edit rates** of all Assets must be equal and constant for the duration of the Composition._

- mpeg_ii_pack_constraints_2v5.doc section 7.3
_The following picture-related parameters shall not change for the duration of the Composition Playlist:
 -- **Raster size**
 -- **Screen Aspect Ratio**
 -- **Frame Rate**_

- mpeg_ii_pack_constraints_2v5.doc section 7.4
_All sound track files referenced by a given composition must share identical:
 -- **baseband sample rate**
 -- **baseband sample bit depth**
 -- **language**
 -- **channel format** (presentation format)_

- mpeg_ii_pack_constraints_2v5.doc section 7.5
All subtitle tracks referenced by a given Composition Playlist must share identical **language**.

**Comment by matmat** [comment]
Hmm.. Maybe keep this check but default to a warning? Dolby DSS release notes for v4.7.3 (2014) contains:

_"DC-12355: Fixed a problem where the DSP100 failed to play certain Digital Cinema Packages (DCPs) containing both encrypted and unencrypted reels."_

**Comment by remia** [comment]
Yes, this is not per spec or standard but rather a compatibility issue with player in the fields. I guess this warrant a comment in the check description. We had quite a few DCP that refused to play in the past because of this and iirc. earlier this year a case of failed playback for an encrypted DCP with un-encrypted Atmos track.

You could always create a custom profile and tag this as a warning (or we should update the default one and use a custom profile internally...). A current limitation is that this specific check is for a list of coherence check and you can not update warning level per key, this needs to be updated.
